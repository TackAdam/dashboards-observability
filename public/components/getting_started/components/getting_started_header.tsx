/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiLink,
  EuiPageHeader,
  EuiPageHeaderSection,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';
import { OPENSEARCH_S3_DOCUMENTATION_URL } from '../../../../common/constants/data_connections';

export const GettingStartedConnectionsHeader = () => {
  return (
    <div>
      <EuiPageHeader>
        <EuiPageHeaderSection>
          <EuiTitle size="l" data-test-subj="gettingstarted-header">
            <h1>Getting Started</h1>
          </EuiTitle>
        </EuiPageHeaderSection>
      </EuiPageHeader>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        Please select your data source.{' '}
        <EuiLink external={true} href={OPENSEARCH_S3_DOCUMENTATION_URL} target="blank">
          Learn more
        </EuiLink>
      </EuiText>
      <EuiSpacer size="l" />
      <EuiSpacer size="s" />
    </div>
  );
};
