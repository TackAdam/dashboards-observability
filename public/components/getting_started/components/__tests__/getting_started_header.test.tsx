/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { GettingStartedConnectionsHeader } from '../getting_started_header';

describe('Getting Started Header Test', () => {
  configure({ adapter: new Adapter() });

  it('Renders getting started header as expected', async () => {
    const wrapper = mount(<GettingStartedConnectionsHeader />);

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
