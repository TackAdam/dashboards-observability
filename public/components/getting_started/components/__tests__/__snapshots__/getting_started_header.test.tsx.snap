// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting Started Header Test Renders getting started header as expected 1`] = `
<GettingStartedConnectionsHeader>
  <div>
    <EuiPageHeader>
      <header
        className="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
      >
        <EuiPageHeaderSection>
          <div
            className="euiPageHeaderSection"
          >
            <EuiTitle
              data-test-subj="gettingstarted-header"
              size="l"
            >
              <h1
                className="euiTitle euiTitle--large"
                data-test-subj="gettingstarted-header"
              >
                Getting Started
              </h1>
            </EuiTitle>
          </div>
        </EuiPageHeaderSection>
      </header>
    </EuiPageHeader>
  </div>
</GettingStartedConnectionsHeader>
`;
